src/tir/analysis/control_flow_graph.cc

```diff
@@ -473,6 +473,7 @@ class ControlFlowGraphBuilder final : public IRVisitorWithAnalyzer {
   void VisitAccess(const BufferAccess& node, BufferTouch::AccessType touch_type,
                    PrimExpr known_value_expr) {
     auto& current_block = out_->control_flow_.back();
+    current_block.current_predicate = CurrentScopePredicate();
```

Comment:
Nitpick: Instead of adding a member variable, can we add this as an additional parameter to `current_block.MakeBufferTouch`?  That way, we avoid accidental re-use of `current_predicate` from other callsites.

Reply:
Thanks! I added an additional parameter to MakeBufferTouch.